src/compiler/checker.ts

```diff
@@ -1412,15 +1415,17 @@ namespace ts {
                     }
                     return undefined;
                 }
-
-                // Create a new symbol to represent the untyped module and store it in globals.
-                // This provides a name to the module. See the test tests/cases/fourslash/untypedModuleImport.ts
-                const newSymbol = createSymbol(SymbolFlags.ValueModule, quotedName);
-                // Module symbols are expected to have 'exports', although since this is an untyped module it can be empty.
-                newSymbol.exports = createMap<Symbol>();
-                // Cache it so subsequent accesses will return the same module.
-                globals[quotedName] = newSymbol;
-                return newSymbol;
+                else {
+                    // Create a new symbol to represent the untyped module and store it in globals.
+                    // This provides a name to the module. See the test tests/cases/fourslash/untypedModuleImport.ts
+                    const newSymbol = createSymbol(SymbolFlags.ValueModule, quotedName);

```

Comment:
I'm still not convinced why do we need to create a new symbol here an stick it into `globals`? AFAIR returning `undefined` will be should be properly interpreted downstream and converted into `unknownSymbol`